src/main/frontend/pipeline-console-view/pipeline-console/main/hooks/use-steps-poller.ts

```diff
@@ -124,56 +124,65 @@ export function useStepsPoller(props: RunPollerProps) {
-
-          if (defaultStep.stageId) {
-            setExpandedSteps((prev) => [...prev, defaultStep.id]);
-            updateStepConsoleOffset(defaultStep.id, false, 0 - LOG_FETCH_SIZE);
+    getRunSteps()
+      .then((steps) => {
+        steps = steps || [];
+        setSteps(steps);
+
+        const usedUrl = parseUrlParams(steps);
+        if (!usedUrl) {
+          const defaultStep = getDefaultSelectedStep(steps);
+          if (defaultStep) {
+            setOpenStage(defaultStep.stageId);
+
+            if (defaultStep.stageId) {
+              setExpandedSteps((prev) => [...prev, defaultStep.id]);
+              updateStepConsoleOffset(
+                defaultStep.id,
+                false,
+                0 - LOG_FETCH_SIZE,
+              );
+            }
           }
         }
-      }
 
-      if (!run?.complete) {
-        startPollingPipeline({
-          getStateUpdateFn: getRunSteps,
-          onData: (data) => {
-            const hasNewSteps =
-              JSON.stringify(stepsRef.current) !== JSON.stringify(data);
-
-            if (userManuallySetNode) {
-              const defaultStep = getDefaultSelectedStep(steps);
-              if (defaultStep) {
-                setOpenStage(defaultStep.stageId);
-
-                if (defaultStep.stageId) {
-                  setExpandedSteps((prev) => [...prev, defaultStep.id]);
-                  updateStepConsoleOffset(
-                    defaultStep.id,
-                    false,
-                    0 - LOG_FETCH_SIZE,
-                  );
+        if (!run?.complete) {
+          startPollingPipeline({
+            getStateUpdateFn: getRunSteps,
+            onData: (data) => {
+              const hasNewSteps =
+                JSON.stringify(stepsRef.current) !== JSON.stringify(data);
+
+              if (userManuallySetNode) {
+                const defaultStep = getDefaultSelectedStep(steps);
+                if (defaultStep) {
+                  setOpenStage(defaultStep.stageId);
+
+                  if (defaultStep.stageId) {
+                    setExpandedSteps((prev) => [...prev, defaultStep.id]);
+                    updateStepConsoleOffset(
+                      defaultStep.id,
+                      false,
+                      0 - LOG_FETCH_SIZE,
+                    );
+                  }
                 }
               }
-            }
 
-            if (hasNewSteps) {
-              setSteps(data);
-              stepsRef.current = data;
-            }
-          },
-          checkComplete: () => !run?.complete,
-          interval: POLL_INTERVAL,
-        });
-      }
-    });
+              if (hasNewSteps) {
+                setSteps(data);
+                stepsRef.current = data;
+              }
+            },
+            checkComplete: () => !run?.complete,
+            interval: POLL_INTERVAL,
+          });
+        }
+        return null;
+      })
+      .catch((error) => {
+        console.error("Error in getRunSteps:", error);
```

Comment:
A shout out for this one. Do you think it can cause unintended side-effects?

Reply:
no should be fine